article-analyzer/src/main/java/com/reader/article_analyzer/JsonFileService.java

```diff
@@ -0,0 +1,39 @@
+package com.reader.article_analyzer;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Service;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Set;
+
+@Service
+public class JsonFileService {
+
+    public <T> List<T> readJsonFile(String filePath, Class<T> classOfT) {
+        try {
+            Resource resource = new ClassPathResource(filePath); // Convert filePath to a string
+            InputStreamReader reader = new InputStreamReader(resource.getInputStream());
+            Type type = TypeToken.getParameterized(List.class, classOfT).getType();
+            return new Gson().fromJson(reader, type);
```

Comment:
nice method, but do not create every time Gson instance, make it as a bean (in config clas) and inject it here

Reply:
Changed it with the creation new class AnalyzerConfiguration